crates/ark-metadata/src/metadata_manager.rs

```diff
@@ -171,16 +171,27 @@ impl<'a, T: Storage, C: StarknetClient, F: FileManager> MetadataManager<'a, T, C
                     is_cache_updated: false,
                 })
             }
-            _ => {
-                debug!("Fetching image... {}", url);
-                let response = reqwest::get(url).await?;
+            ImageCacheOption::Save => {
+                let response = self
+                    .request_client
+                    .get(url)
+                    .timeout(Duration::from_secs(5))
```

Comment:
I would use it as an argument to ensure we can change this later with very few changes.